src/vcl/ImageData.cc

```diff
@@ -318,26 +330,34 @@ ImageData::ImageData(void* buffer, cv::Size dimensions, int cv_type)
     _tdb->set_compression(_compress);
 }
 
-ImageData::ImageData(const ImageData &img)
+ImageData::ImageData(const ImageData &img, bool copy)
 {
     _format = img._format;
     _compress = img._compress;
     _image_id = img._image_id;
 
-    if ( !(img._cv_img).empty() )
-        copy_cv(img._cv_img);
+    if ( !(img._cv_img).empty() ) {
+        if (copy) {
+            deep_copy_cv(img._cv_img);
+        }
+        else {
+            shallow_copy_cv(img._cv_img);
+        }
+    }
 
     if ( img._tdb != NULL )
         _tdb = new TDBImage(*img._tdb);
     else
         _tdb = NULL;
 
     int start;
+    // TODO IS THIS REALLY NEEDED????
```

Comment:
Its better to not have comments like that. Either you can resolve it or file an issue and remove the comment